jquery.xmleditor.js

```diff
@@ -4856,6 +4888,202 @@ XMLElementStub.prototype.isSelected = function() {
+XMLTemplates.prototype.templateForm = function() {
+    var form = '<div class="template-form" title="Choose Template">' +
+        '<ul>';
+
+    for(var i=0; i<this.templates.length; i++) {
+        var current = this.templates[i];
+
+        form += '<li class="templating"'  + ' id="template_' + i + '">' +
+            '<a href="' + current.filename + '">';
+
+        if (current.icon_url) {
+           form += '<img class="' + current.icon_class + '" src="' + current.icon_url + '"/> ';
+        }
+
+        if (current.icon_class) {
+           form += '<i class="' + current.icon_class + '"></i> ';
+        }
+        form += '<div>';
+        form += this._formatFormText(current.filename);
+
+        if (current.description) {
+            form += '<span>' + current.description + '</span>';
+        }
+
+        form += '</div>';
+
+        form += '</a>' +
+            '</li>';
+    }
+
+    form += '</ul>' +
+        '</div>';
+
+    this.form = $(form);
+    this.form.insertAfter("body");
+    $('li:first', this.form).addClass('focus');
+};
+
+/**
+ * Select a template from the form
+ * @param dialog
+ * @param self
+ */
+XMLTemplates.prototype.processForm = function() {
+    // Split on mdash if description present
+    var selection = $(".focus a", this.form).attr('href');
+
+    this.form.dialog("close");
+    this.loadSelectedTemplate(selection);
+};
+
+/**
+ * Load selected template.
+ * @param selection
+ * @param self
+ */
+XMLTemplates.prototype.loadSelectedTemplate = function(selection) {
+    var self = this;
+    // Default template loading doesn't have access to xml_templates constructor
+    $.ajax({
+        url: this.template_path + selection,
+        dataType: "xml"
+    }).done(function(data) {
+        var xml_string = self.editor.xml2Str(data);
+        self.editor._documentReady(xml_string);
+        self.editor.loadSchema(self.editor.options.schema);
+    }).fail(function(jqXHR, textStatus) {
+        self.editor.loadSchema(self.editor.options.schema);
+        alert("Unable to load the requested template: " + textStatus);
+    });
+};
+
+/**
+ * Highlight and focus currently selected template
+ * If enter hit go ahead and load focused template
+ * @param dialog
+ */
+XMLTemplates.prototype.focusTemplate = function() {
+    var self = this;
+
+    // Focus selected template
+    this.form.on('keydown click', '.templating', function(e) {
+
+        e.preventDefault();
+        var key = e.which;
+        var number_of_forms, base_element, current, form_id, next_element;
+
+        if (key === 1 || key === 9) {
+            // Left click, select the clicked target
+            if (key == 1) {
+                base_element = $(e.target);
+                if (!base_element.hasClass("templating")) {
+                    base_element = base_element.parent(".templating");
+                }
+            } else {
+                // Tab, select the next template
+                current = $('.focus', self.form).attr('id');
```

Comment:
Tab selection isn't working a lot of the time for me.  Right after loading the editor it doesn't work, and i seem to have to click on a number of templates before it starts working.  Also, if you tab past the end of the list it throws an exception.

I'd recommend trying to use the jquery next function to find the next .templating element.  If the user is at the end of the list of templates, it should probably cycle back to the beginning since there is no going backwards